pkg/codegen/schema/schema.go

```diff
@@ -2023,6 +2033,9 @@ func bindDefaultValue(path string, value interface{}, spec *DefaultSpec, typ Typ
 		for name, raw := range spec.Language {
 			language[name] = json.RawMessage(raw)
 		}
+		if spec.Environment == nil {
+			diags = diags.Append(errorf(path, "Default must specify an environment"))
+		}
```

Comment:
This is not correct: it is valid for defaults to provide only static values and no env vars. It seems reasonable to require one or the other, though.

Reply:
I may have read this wrong. [The schema docs](https://www.pulumi.com/docs/guides/pulumi-packages/schema/) is a little confusing.

Reply:
Ah, no, this is my mistake. We spread default information out into a couple of places, and if a `DefaultSpec` is provided, then the `Environment` field should be non-empty. With that in mind, I think the rule should be 

```suggestion
		if len(spec.Environment) == 0 {
			diags = diags.Append(errorf(path, "default must specify an environment"))
		}
```